### PR TITLE
Improve font options. Add system fonts selector and font size slider.

### PR DESCRIPTION
**Pull request by jacobbuck**
I've also namespaced the built-in fonts with `'Sight '` to prevent conflicts with system fonts.

### CODE REVIEW

**Comment by deepfriedmind** [comment]
This sounds great. I hope this gets merged soon.

**Comment by jacobbuck** [comment]
@tsenart :wave:

File: js/background.js [hunk 1]
```diff
@@ -49,7 +49,7 @@
   const OPTIONS_DEFAULTS = {
     theme: 'sunburst',
     font: 'Inconsolata',
-    fontSize: 'medium',
+    fontSize: 16,
```
Comment (tsenart) [inline js/background.js]: `16` what? I'd rather have the unit go with the value.

File: css/main.css [hunk 2]
```diff
@@ -1,68 +1,68 @@
 @font-face {
-  font-family: 'Inconsolata';
+  font-family: 'Sight Inconsolata';
```
Comment (tsenart) [inline css/main.css]: Why prefixing all of these font names with `Sight`?

File: js/options.js [hunk 3]
```diff
@@ -31,7 +32,7 @@
       selector: '#font-size',
       value: 'value',
       decode: id,
-      render: set('#code', 'style.fontSize', id)
+      render: set('#code', 'style.fontSize', px)
```
Comment (tsenart) [inline js/options.js]: As mentioned before, I'd rather have the unit come with the value, so we could use id here, no?

File: js/options.js [hunk 4]
```diff
@@ -47,6 +48,21 @@
   };
 
   doc.addEventListener('DOMContentLoaded', function() {
+    var fontOpt = localStorage.getItem('font');
+    var fontEl = doc.querySelector('#font');
+    chrome.fontSettings.getFontList(function (fonts) {
```
Comment (tsenart) [inline js/options.js]: No space between function and left paren please.

File: js/options.js [hunk 5]
```diff
@@ -47,6 +48,21 @@
...
+      var frag = doc.createDocumentFragment();
+      fonts.forEach(function (font) {
+        var opt = doc.createElement('option');
+        opt.textContent = font.displayName;
+        opt.value = font.fontId;
+        frag.appendChild(opt);
+      });
+      doc.querySelector('#font-system').appendChild(frag);
+      fontEl.value = fontOpt;
+      fontEl.dispatchEvent(new Event('change'));
```
Comment (tsenart) [inline js/options.js]: These two lines aren't necessary I think, because we're doing this below already.

**Comment by tsenart** [comment]
@jacobbuck: PTAL

File: js/background.js (hunk 1 above)
Comment (jacobbuck) [inline js/background.js]: Using font sizes as int to easily manipulate. `16` becomes `16px`. Could use `em`s if preferred.

File: css/main.css (hunk 2 above)
Comment (jacobbuck) [inline css/main.css]: To prevent conflicts with system installed fonts.

File: js/options.js (hunk 5 above)
Comment (jacobbuck) [inline js/options.js]: Unfortunately they are as `chrome.fontSettings.getFontList` is async, otherwise you get a race condition :car:

**Comment by jacobbuck** [comment]
@tsenart I've responded to a few comments. Will change the rest when I've got a moment.

**Comment by tsenart** [comment]
@jacobbuck: Feel free to re-open when you got the moment :-)

**Pull request closed without merging**